locust/test/test_main.py

```diff
@@ -102,29 +103,30 @@ def my_task(self):
             )
         ) as file_path:
             # print(subprocess.check_output(["cat", file_path]))
-            proc = subprocess.Popen(
-                ["locust", "-f", file_path, "--custom-string-arg", "command_line_value", "--web-port", str(port)],
-                stdout=PIPE,
-                stderr=PIPE,
-                text=True,
-            )
-            gevent.sleep(1)
-
-        requests.post(
-            f"http://127.0.0.1:{port}/swarm",
-            data={"user_count": 1, "spawn_rate": 1, "host": "https://localhost", "custom_string_arg": "web_form_value"},
-        )
-        gevent.sleep(1)
-
-        proc.send_signal(signal.SIGTERM)
-        stdout, stderr = proc.communicate(timeout=3)
-        self.assertIn("Starting Locust", stderr)
-        self.assertRegex(stderr, r".*Shutting down[\S\s]*Aggregated.*", "no stats table printed after shutting down")
-        self.assertNotRegex(stderr, r".*Aggregated[\S\s]*Shutting down.*", "stats table printed BEFORE shutting down")
-        self.assertNotIn("command_line_value", stdout)
-        self.assertIn("web_form_value", stdout)
+            with TestProcess(
+                f"locust -f {file_path} --custom-string-arg command_line_value --web-port {port}",
+                self.fail,
```

Comment:
Can we have this as the default? If you have a hard time getting hold of the testcase instance (self) inside TestProcess, maybe try pytest.fail instead?